### PR TITLE
[CONSVC-1970] test: fix remote_settings__refresh failure

### PR DESCRIPTION
**Pull request by Trinaa**
## Description
* fix for flake issue in `remote_settings__refresh`
* suspect that the issue was caused by using different record IDs when posting attachments
* verified for flake by executing tests 100x consecutively on local and [CI](https://app.circleci.com/pipelines/github/mozilla-services/merino-py/377/workflows/19cdaad0-1251-436d-a23a-2c08cde488a7/jobs/1209) without error

## Issue(s)
[CONSVC-1970](https://mozilla-hub.atlassian.net/browse/CONSVC-1970)

### CODE REVIEW

**Review by pjenvey — APPROVED** [review]

**Review by ncloudioj — APPROVED** [review]
Nice detective work 🕵️‍♀️

File: tests/contract/client/tests/test_merino.py [hunk 1]
```diff
@@ -59,13 +60,13 @@ def kinto_step(step: Step) -> None:
                 f"Unsupported request type {type(step.request)} for Kinto service step."
             )
 
-        record: KintoRequestRecord = kinto_records[step.request.filename]
+        record: KintoRequestAttachment = kinto_records[step.request.filename]
```
Comment (hackebrot) [inline tests/contract/client/tests/test_merino.py:63]:
```suggestion
        attachment: KintoRequestAttachment = kinto_attachments[step.request.filename]
```

File: tests/contract/client/tests/conftest.py [hunk 2]
```diff
@@ -43,27 +42,24 @@ def fixture_kinto_environment(request: Any) -> KintoEnvironment:
 
 
 @pytest.fixture(scope="session", name="kinto_records")
-def fixture_kinto_records(request: Any) -> dict[str, KintoRequestRecord]:
+def fixture_kinto_records(request: Any) -> dict[str, KintoRequestAttachment]:
```
Comment (hackebrot) [inline tests/contract/client/tests/conftest.py:45]:
```suggestion
def fixture_kinto_attachments(request: Any) -> dict[str, KintoRequestAttachment]:
```

File: tests/contract/client/tests/conftest.py [hunk 3]
```diff
@@ -43,27 +42,24 @@ def fixture_kinto_environment(request: Any) -> KintoEnvironment:
...
         ]
-        kinto_records[data_file.name] = KintoRequestRecord(
-            id=data_file.stem,
-            attachment=KintoRequestAttachment(
-                filename=data_file.name,
-                filecontent=content,
-                mimetype="application/json",
-                suggestions=suggestions,
-            ),
+        kinto_records[data_file.name] = KintoRequestAttachment(
```
Comment (hackebrot) [inline tests/contract/client/tests/conftest.py:58]: See above

File: tests/contract/client/tests/kinto.py [hunk 4]
```diff
@@ -107,25 +96,24 @@ def get_record(environment: KintoEnvironment, record_id: str) -> KintoResponseRe
 
 
 def upload_attachment(
-    environment: KintoEnvironment, record: KintoRequestRecord, data_type: str
+    environment: KintoEnvironment,
+    record_id: str,
+    record: KintoRequestAttachment,
```
Comment (hackebrot) [inline tests/contract/client/tests/kinto.py:101]:
```suggestion
    attachment: KintoRequestAttachment,
```

**Review by hackebrot — CHANGES_REQUESTED** [review]
Hi @Trinaa! 👋🏻 

Awesome work!

For the sake of consistency and ergonomics, I recommend updating fixture and variable names to match the types that they return and hold: `record -> attachment`.

**Comment by Trinaa** [comment]
Hi @hackebrot 👋🏻,

I agree the `attachment` makes more sense than `record`. Code should be ready for another look!

**Review by hackebrot — APPROVED** [review]
LGTM! 🥳

**Pull request merged**